safe-discriminant-derive/src/lib.rs

```diff
@@ -80,13 +80,30 @@ fn validate_all_variants(variants: impl Iterator<Item = Variant>) -> Result<()>
         .unwrap_or(Ok(()))
 }
 
+/// Returns true if there is any #[x] where x is not `repr`
+/// returns false otherwise.
+fn contains_attribute_macros(attrs: &[Attribute]) -> bool {
+    attrs
+        .iter()
+        .filter(|attr| !attr.path().is_ident("repr"))
+        .count()
+        != 0
+}
 /// Constructs Discriminant trait implementation for given enum.
 /// Returns error in one of two cases:
 /// 1- No valid `#[repr(x)]` is found.
 /// 2- Any of the enum variants is missing discriminant.
+/// 3- contains any additional top level attribute macros.
 fn derive_discriminant_inner(tagged_enum: ItemEnum) -> Result<TokenStream> {
     let prim = get_enum_repr_prim(&tagged_enum.attrs, tagged_enum.ident.span())?;
     validate_all_variants(tagged_enum.variants.into_iter())?;
+    if contains_attribute_macros(&tagged_enum.attrs) {
+        return Err(Error::new(
+            tagged_enum.ident.span(),
+            "Discriminant is not compatiable with any top \
```

Comment:
Nit: I'm still not totally sure how I feel about these `\` continuations here. It seems like that will create a string like this: "Discriminant is not compatiable with any top <lots of spaces here> level `#[attr]` except `#[repr(_)]`."

Maybe the compiler error is reformatted so the end user can't see this, so probably not a huge deal, but `concat!` is how I thought you were supposed to do string literals across multiple lines: https://doc.rust-lang.org/std/macro.concat.html

Reply:
They do the same, I am used to the ` \` notation because it is similar to C, but we can standardize using `concat!`